kolibri/core/assets/src/views/k-radio-button.vue

```diff
@@ -1,47 +1,49 @@
 <template>
 
-  <div
-    class="k-radio-container"
-    :class="{ 'k-radio-disabled': disabled }"
-    @click="select"
-  >
-    <div class="tr">
-
-      <div class="k-radio" :class="{ 'k-radio-active': isActive }">
-        <input
-          ref="kRadioInput"
-          type="radio"
-          class="k-radio-input"
-          :id="id"
-          :value="radiovalue"
-          :disabled="disabled"
-          :autofocus="autofocus"
-          @focus="isActive = true"
-          @blur="isActive = false"
-          @change="emitChange"
-          v-model="model"
-          @click.stop="select"
-        >
-
-        <mat-svg
-          v-if="isCurrentlySelected"
-          category="toggle"
-          name="radio_button_checked"
-          class="k-radio-selected"
-        />
-        <mat-svg
-          v-else
-          category="toggle"
-          name="radio_button_unchecked"
-          class="k-radio-unselected"
-        />
-
-      </div>
-
-      <label :for="id" class="k-radio-label" @click.prevent>{{ label }}</label>
-
-    </div>
-  </div>
+  <!-- HTML makes clicking label apply to input by default -->
+  <label class="k-radio-button">
+    <!-- v-model listens for @input event by default -->
+    <!-- @input has compatibility issues for input of type radio -->
+    <!-- Here, manually listen for @change (no compatibility issues) -->
+    <input
+      ref="input"
+      type="radio"
+      class="input"
+      :id="id"
+      :checked="isChecked"
+      :value="radiovalue"
+      :disabled="disabled"
+      :autofocus="autofocus"
+      @focus="active = true"
+      @blur="active = false"
+      @change="update($event)"
+    >
+    <!-- the radio buttons the user sees -->
+    <mat-svg
+      v-if="isChecked"
+      category="toggle"
+      name="radio_button_checked"
+      :class="['checked', {disabled, active}]"
+    />
+    <mat-svg
+      v-else
+      category="toggle"
+      name="radio_button_unchecked"
+      :class="['unchecked', {disabled, active}]"
+    />
+
+    <span :class="['text', {disabled}]">
+      {{ label }}
+      <span
+        v-if="description"
+        :class="['description', {disabled}]"
+      >
+        <br>
```

Comment:
minor note - don't need to change...

might have been better if the description and label were implemented as two block-level `div`s

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/br

Reply:
This was the case in my first few iterations. Wanted to avoid block level because I wanted <label> to be the top level tag and Block level tags within <label> aren’t valid HTML. 

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

I had display:block on a few of the  components as well, but remembered the discussion we had about mixing online and block elements, opting for this approach. 

Doing it this way ^ is the quickest way to get rid of the <br>

Tried using k-grid too, but this seemed simpler after implementing it. Also couldn’t define anything other than a <div>, which brought me back to the first point.

Reply:
Note that using multiple block-level elements might also help with the line-height issue I mentioned below, because you can give each one its own line height.

> I had display:block on a few of the tags as well, but remembered the discussion we had about mixing online and block elements, opting for this approach.

A few `span`s with `display: block` seems like a reasonable workaround if we're getting validation errors putting `div`s inside the `label`. (I didn't know that was the case!)

But yes, within a single parent, all children should either be inline or block-level - not mixed.

> Tried using k-grid 

agreed, `k-grid` doesn't seem like the solution here

Reply:
> ..reasonable workaround if we're getting validation errors putting divs inside the label. (I didn't know that was the case!)
It actually isn't! There aren't any build error or anything, it's just not HTML spec-compliant.